src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+            # the mesh's vertical center.
+            vertical_bounds = (self.mesh.bounds[0][2], self.mesh.bounds[1][2])
+            mesh_height = vertical_bounds[1] - vertical_bounds[0] + 1
+            centerZ = (vertical_bounds[1] + vertical_bounds[0])/2
+
+            # Compute the bounded footprint and recursively compute the intersection
+            bounded_footprint = other.approxBoundFootprint(centerZ, mesh_height)
+
+            return self.difference(bounded_footprint)
+
+        # Don't know how to compute this difference, fall back to default behavior.
+        return super().difference(other)
+
+
+    def uniformPointInner(self):
+        # TODO: Look into tetrahedralization, perhaps to be turned on when a heuristic
+        # is met. Currently using Trimesh's rejection sampling.
+        sample = trimesh.sample.volume_mesh(self.mesh, self.num_samples)
+
+        if len(sample) == 0:
+            raise RejectionException("Rejection sampling MeshVolumeRegion failed.")
+        else:
+            return Vector(*sample[0])
+
+    def distanceTo(self, point):
+        """Get the minimum distance from this region to the specified point."""
+        point = toVector(point, "Could not convert 'point' to vector.")
+
+        pq = trimesh.proximity.ProximityQuery(self.mesh)
+
+        dist = pq.signed_distance([point.coordinates])[0]
+
+        # Positive distance indicates being contained in the mesh.
+        if dist > 0:
+            dist = 0
+
+        return abs(dist)
+
+    @cached_property
+    def inradius(self):
+        assert not needsSampling(self)
+        center_point = self.mesh.bounding_box.center_mass
+
+        pq = trimesh.proximity.ProximityQuery(self.mesh)
+        region_distance = abs(pq.signed_distance([center_point])[0])
+
+        if region_distance < 0:
+            return 0
+        else:
+            return region_distance
+
+    @property
+    def dimensionality(self):
+        return 3
+
+    @cached_property
+    def size(self):
+        return self.mesh.mass/self.mesh.density
+
+    ## Utility Methods ##
+    @lru_cache(maxsize=None)
+    def getSurfaceRegion(self):
+        """ Return a region equivalent to this one, except as a MeshSurfaceRegion"""
+        return MeshSurfaceRegion(self.mesh, self.name, orientation=self.orientation, \
+            tolerance=self.tolerance, centerMesh=False, onDirection=self.onDirection, engine=self.engine)
+
+    def getVolumeRegion(self):
+        """ Returns this object, as it is already a MeshVolumeRegion"""
+        return self
+
+
+class MeshSurfaceRegion(MeshRegion):
+    """ An instance of MeshRegion that performs operations over the surface of the mesh.
+
+    If an orientation is not passed to this mesh, a default orientation is
+    provided which provides an orientation that aligns an instance's z axis 
+    with the normal vector of the face containing that point.
```

Comment:
Should specify which of the infinitely many orientations consistent with this description is chosen.

Reply:
I'm actually not sure, since we use Trimesh's function for it.

Reply:
We now ensure the orientation has yaw aligned with a global yaw of 0.